src/content_scripts/applyColorSchemeJs.js

```diff
@@ -136,15 +368,70 @@ function matchMediaOverwrite(...args) {
 function applyJsOverwrite() {
     // do not overwrite twice
     if (overwroteMatchMedia) {
+        dispatchChangeEvents();
         return;
     }
 
-    // add hint for websites to detect this has been faked
-    // matchMediaOverwrite.FAKED = ADDON_FAKED_WARNING;
-    // DISABLED, as it could be used for tracking (which simple console statements itself cannot AFAIK)
+    if (fakedColorStatus === COLOR_STATUS.NO_OVERWRITE) {
+        // eslint-disable-next-line no-global-assign
+        jsLastColorStatus = getSystemMediaStatus();
+    } else {
+        // eslint-disable-next-line no-global-assign
+        jsLastColorStatus = fakedColorStatus;
+    }
+
+    // actually overwrite
+
+    Reflect.defineProperty(MediaQueryListPrototype, 'addListener', {
+        configurable: true,
+        enumerable: true,
+        value: exportFunction(skeleton.addListener, window),
+        writable: true
+    });
+    Reflect.defineProperty(MediaQueryListPrototype, 'removeListener', {
+        configurable: true,
+        enumerable: true,
+        value: exportFunction(skeleton.removeListener, window),
+        writable: true
+    });
+    let descriptorMatches = Reflect.getOwnPropertyDescriptor(skeleton, 'matches');
+    Reflect.defineProperty(MediaQueryListPrototype, 'matches', {
+        configurable: true,
+        enumerable: true,
+        get: exportFunction(descriptorMatches.get, window)
+    });
+    let descriptorOnchange = Reflect.getOwnPropertyDescriptor(skeleton, 'onchange');
+    Reflect.defineProperty(MediaQueryListPrototype, 'onchange', {
+        configurable: true,
+        enumerable: true,
+        get: exportFunction(descriptorOnchange.get, window),
+        set: exportFunction(descriptorOnchange.set, window),
+    });
+
+    Reflect.defineProperty(EventTargetPrototype, 'addEventListener', {
+        configurable: true,
+        enumerable: true,
+        value: exportFunction(skeleton.addEventListener, window),
+        writable: true
+    });
+    Reflect.defineProperty(EventTargetPrototype, 'removeEventListener', {
+        configurable: true,
+        enumerable: true,
+        value: exportFunction(skeleton.removeEventListener, window),
+        writable: true
+    });
 
-    // actually overwrite function
-    exportFunction(matchMediaOverwrite, window, {defineAs: "matchMedia"});
+    overwroteMatchMedia = true;
 }
 
 applyJsOverwrite();
+
+/**
+ * Logs ADDON_FAKED_WARNING if not already did
+ */
```

Comment:
+ `@private` 

And AFAIK you also need to set `@returns {void}`. Please have a look at the EsLint errors with our config, they should tell you.